richgo.go

```diff
@@ -30,27 +32,25 @@ func main() {
 	case 1:
 		cmd = exec.Command("go")
 	default:
-		cmd = exec.Command("go", os.Args[1:]...)
-		// select a wrapper with subcommand
-		if f, ok := lps[os.Args[1]]; ok {
-			factory = f
+		// This is a bit of a special case. Somebody is already
+		// running `go test` for us, and just wants us to prettify the
+		// output.
+		if os.Args[1] == testFilterCmd {
+			cmd = exec.Command("cat", "-")
+			factory = test.New
+		} else {
+			cmd = exec.Command("go", os.Args[1:]...)
+			// select a wrapper with subcommand
+			if f, ok := lps[os.Args[1]]; ok {
+				factory = f
+			}
 		}
 	}
 
-	var stderr io.WriteCloser
-	if editor.Formattable(os.Stderr) {
-		stderr = editor.Stream(os.Stderr, factory())
-	} else {
-		stderr = editor.Stream(os.Stderr, editor.Parrot())
-	}
+	stderr := formatWriteCloser(os.Stderr, factory)
```

Comment:
Let me know if you'd prefer I split this up in a separate PR. I saw it and felt compelled to de-dupe it, hope you don't mind.